### PR TITLE
Updated docs on integrating with various operating system features

### PR DESCRIPTION
**Pull request by tleonhardt**
Updated docs on integrating with various operating system features

Also:
- Moved a couple sub-sections from "Miscellaneous" features to "Integrating with OS"

This partially addresses #765

### CODE REVIEW

File: docs/features/os.rst [hunk 1]
```diff
@@ -1,14 +1,60 @@
...
+operating system shell::
+
+    (Cmd) shell ls -al
+
+If you use the default :ref:`features/shortcuts_aliases_macros:Shortcuts`
+defined in ``cmd2`` you'll get a ``!`` shortcut for ``shell``, which allows you
+to type::
+
+    (Cmd) !ls -al
+
```
Comment (kmvanbrunt) [inline docs/features/os.rst]: Maybe add something about shell commands and their arguments tab completing

File: docs/features/os.rst [hunk 2]
```diff
@@ -1,14 +1,60 @@
...
+
+Editors
+-------
+
+``cmd2`` includes the built-in ``edit`` command which uns a text editor and
```
Comment (kmvanbrunt) [inline docs/features/os.rst]: Change uns to runs

File: docs/features/os.rst [hunk 3]
```diff
@@ -1,14 +1,60 @@
...
+optionally open a file with it::
```
Comment (kmvanbrunt) [inline docs/features/os.rst]: Change open to opens

File: docs/features/os.rst [hunk 4]
```diff
@@ -1,14 +1,60 @@
...
+for any in a list of common editors for your operating system.
+
+Terminal pagers
+---------------
+
+Output of any command can be displayed one page at a time using the
+:meth:`~.cmd2.Cmd.ppaged` method.
+
+Alternatively, a terminal pager can be invoked directly using the ability
+to run shell commands with the ``!`` shortcut.
```
Comment (kmvanbrunt) [inline docs/features/os.rst]: Perhaps add an example: `!less`

File: docs/features/os.rst [hunk 5]
```diff
@@ -84,3 +130,37 @@ your own argument parsing of the command line::
...
+Automating cmd2 apps from other CLI/CLU tools
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+While ``cmd2`` is designed to create **interactive** command-line applications
+which enter a Read-Evaluate-Print-Loop (REPL), there are a great many times
+when it would be useful to use a ``cmd2`` application as a run-and-done
+command-line utility for purposes of automation and scripting.
+
+This is easily achieved by combining the following capabilities of ``cmd2``:
+1. Ability to invoke a ``cmd2`` application with arguments
```
Comment (kmvanbrunt) [inline docs/features/os.rst]: Should these all be numbered 1?

File: docs/features/os.rst (hunk 1 above)
Comment (tleonhardt) [reply docs/features/os.rst]: Good call.

File: docs/features/os.rst (hunk 2 above)
Comment (tleonhardt) [reply docs/features/os.rst]: I wish Sphinx had a built-in spell checker.  There was an add-on, but I think its out of date and unmaintained.

File: docs/features/os.rst (hunk 3 above)
Comment (tleonhardt) [reply docs/features/os.rst]: Will do

File: docs/features/os.rst (hunk 4 above)
Comment (tleonhardt) [reply docs/features/os.rst]: I did and also added a note about how once you are in a pager, the pager has control of the terminal.

File: docs/features/os.rst (hunk 5 above)
Comment (tleonhardt) [reply docs/features/os.rst]: Great catch.  I had my Markdown vs reStructuredText wires crossed.  They should all be `#.` and there should be a blank line separating them from the rest of the text.

**Pull request merged**